src/internal/packager/images/progress.go

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package images provides image-related utilities for Zarf
+package images
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/zarf-dev/zarf/src/pkg/utils"
+	"oras.land/oras-go/v2"
+)
+
+// Report defines a function to log progress
+type Report func(bytesRead, totalBytes int64)
+
+// DefaultReport returns a default report function
+func DefaultReport(l *slog.Logger, msg string, imageName string) Report {
+	return func(bytesRead, totalBytes int64) {
+		percentComplete := float64(bytesRead) / float64(totalBytes) * 100
+		remaining := float64(totalBytes) - float64(bytesRead)
+		l.Info(msg, "name", imageName, "complete", fmt.Sprintf("%.1f%%", percentComplete), "remaining", utils.ByteFormat(remaining, 2))
+	}
+}
+
+const defaultProgressInterval = 5 * time.Second
+
+// StartReporting starts the reporting goroutine
+func (tt *TrackedTarget) StartReporting() {
```

Comment:
Given that the push is governed by the caller-supplied context - I wonder if this should accept context and if we could remove more of the bespoke `stopReports` in favor of a `ctx.Done()`

Reply:
Yeah it should definitely accept a context. Though to get rid of the `StopReporting` call we'd have to make a separate context for `StartReporting` then cancel it. I like the feel of the `defer StopReporting` slightly more, but I don't feel too strongly either way